### DISCUSSION TITLE
Next 13 Upgrade Typescript error

### DISCUSSION THREAD

**Discussion by yudyananda** [i18next/next-i18next#2044] [Q&A]
Follow the [upgrade instruction](https://github.com/i18next/next-i18next/blob/master/UPGRADING.md) but now i have typescript complaining about `Type instantiation is excessively deep and possibly infinite`

![photo_2022-12-03_18-13-36](https://user-images.githubusercontent.com/29258979/205438146-0da85757-25f6-4385-affd-197e9e19444f.jpg)

```
// src/types/next-i18next.d.ts
import 'i18next';
import type common from './src/locales/en/common.json';
import type home from './src/locales/en/home.json';

interface I18nNamespaces {
  common: typeof common;
  home: typeof home;
}

declare module 'i18next' {
  interface CustomTypeOptions {
    resources: I18nNamespaces;
  }
}

// pages/index.ts
export default function HomePage(props: any) {
  const router = useRouter();
  const { t } = useTranslation('common');

  return (
    <div>
      {router.locale} - {t('sapaan')}
    </div>
  );
}
```

any idea?

**Comment by adrai** [comment]
@pedrodurek related? https://github.com/i18next/i18next/issues/1857

**Reply by yudyananda** [reply]
now it's working somehow, but i wonder why the autocomplete only working for the object?

for example vscode doesn't give any autocomplete for something like `{t('nav.${index}.sub.${subIndex}.label')}` it just stuck in the key of array and typescript complaining about `Type ... is not assignable to type 'ReactI18NextChild | Iterable<ReactI18NextChild>` 

```json
{
  "nav": [
    {
      "label": "Insight"
    },
    {
      "label": "Users",
      "sub": [
        {
          "label": "Applicant"
        },
        {
          "label": "Student"
        }
      ]
    }
  ]
}
```

**Reply by adrai** [reply]
https://www.i18next.com/overview/typescript#type-error-template-literal

**Reply by yudyananda** [reply]
`as const` didn't work for me ;(
![i18n](https://user-images.githubusercontent.com/29258979/205890760-d3246df1-ad61-48da-9d45-54d2a0ccd771.png)

**Reply by adrai** [reply]
you may need to provide a minimal reproducible example... it seems to work
